src/main/java/duke/command/MarkingCommand.java

```diff
@@ -0,0 +1,34 @@
+package duke.command;
+
+import duke.dukeexception.DukeException;
+import duke.storage.Storage;
+import duke.tasklist.TaskList;
+
+/**
+ * Represents a command to mark a certain task as done or undone.
+ */
+public class MarkingCommand extends Command {
+    private String fullCommand;
+    /**
+     * Class constructor.Construct a command for marking.
+     * @param fullCommand A string of a line from System.in. Begin with "mark" or "unmark".
+     */
+    public MarkingCommand(String fullCommand) {
+        this.fullCommand = fullCommand;
+    }
+
+    /**
+     * Execute the mark command. Mark the index task as done or undone.
+     * @param taskList The taskList containing the target task.
+     * @param storage The object containing the corresponding file.
+     * @throws DukeException
+     */
+    @Override
+    public void execute(TaskList taskList, Storage storage) throws DukeException {
+        if (this.fullCommand.split(" ")[0].equals("mark")) {
+            taskList.markAsDone(Integer.valueOf(this.fullCommand.split(" ")[1]) - 1, storage);
```

Comment:
Same as above maybe better to split up the steps instead of squeezing them into 1 line. Hard to read. Applies to the line, 2 lines down too.